server/src/main/java/org/opensearch/identity/tokens/AuthToken.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.tokens;
+
+/**
+ * Interface for all token formats to support to authenticate user such as UserName/Password tokens, Access tokens, and more.
+ *
+ * @opensearch.experimental
+ */
+public interface AuthToken {
```

Comment:
I don't think we want to make use of Marker Interfaces unless we have to. Is there a reason this is empty?

Reply:
Why not use a marker interface?  Tokens themselves should not have their string representation used directly, sometimes they will reference a username and password, other times a trusted principal, there isn't direct shared functional but implied functionality.

Reply:
> AuthTokens not need get, set, a constructor

Minimal designs purposeful designs are easier to manage, we are deferring the 'what auth tokens' are supported directly into the Subject implementation where they will have to accept/reject the token if it can be processed.  Maybe a clearer pattern will emerge as we build out more token implementations and we can make revisions to this interface.